packages/react/menu/src/Menu.tsx

```diff
@@ -94,65 +237,179 @@ const MenuContent = React.forwardRef((props, forwardedRef) => {
+  const ref = React.useRef<React.ElementRef<typeof MenuContentImpl>>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
 
-    /**
-     * Event handler called when auto-focusing on close.
-     * Can be prevented.
-     */
-    onCloseAutoFocus?: FocusScopeOwnProps['onUnmountAutoFocus'];
+  // Hide everything from ARIA except the `MenuContent`
+  React.useEffect(() => {
+    const content = ref.current;
+    if (content) return hideOthers(content);
+  }, []);
 
-    /**
-     * Whether scrolling outside the `MenuContent` should be prevented
-     * (default: `false`)
-     */
-    disableOutsideScroll?: boolean;
+  return (
+    <MenuContentImpl
+      {...props}
+      ref={composedRefs}
+      onDismiss={() => context.onOpenChange(false)}
+      onEntryFocus={(event) => event.preventDefault()}
+      onOpenAutoFocus={composeEventHandlers(props.onOpenAutoFocus, (event) => {
+        // explicitly focus the content area only when opening
+        event.preventDefault();
+        ref.current?.focus();
+      })}
+    />
+  );
+}) as MenuRootContentPrimitive;
 
-    /**
-     * The direction of navigation between menu items.
-     * @defaultValue ltr
-     */
-    dir?: RovingFocusGroupOwnProps['dir'];
+/* ---------------------------------------------------------------------------------------------- */
 
-    /**
-     * Whether keyboard navigation should loop around
-     * @defaultValue false
-     */
-    loop?: RovingFocusGroupOwnProps['loop'];
+type MenuSubContentOwnProps = Omit<
+  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  | keyof MenuContentImplPrivateProps
+  | 'side'
+  | 'portalled'
+  | 'disabledOutsidePointerEvents'
+  | 'disableOutsideScroll'
+  | 'trapFocus'
+  | 'onOpenAutoFocus'
+  | 'onCloseAutoFocus'
+>;
 
-    /**
-     * Whether the `MenuContent` should render in a `Portal`
-     * (default: `true`)
-     */
-    portalled?: boolean;
-  }
+type MenuSubContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuSubContentOwnProps
+>;
+
+const MenuSubContent = React.forwardRef((props, forwardedRef) => {
+  const context = useMenuContext(CONTENT_NAME);
+  const ref = React.useRef<MenuContentElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+  return context.isSubmenu ? (
+    <MenuContentImpl
+      align="start"
+      id={context.contentId}
+      {...props}
+      ref={composedRefs}
+      side={context.dir === 'rtl' ? 'left' : 'right'}
+      portalled
+      disableOutsidePointerEvents={false}
+      disableOutsideScroll={false}
+      trapFocus={false}
+      onEntryFocus={context.onEntryFocus}
+      onOpenAutoFocus={(event) => event.preventDefault()}
+      // The menu might close because of focusing another menu item in the parent menu. We
+      // don't want it to refocus the trigger in that case so we handle trigger focus ourselves.
+      onCloseAutoFocus={(event) => event.preventDefault()}
```

Comment:
Do we still want to compose these as I think we used to no?

Reply:
We never auto focus on close so why would they need to bind to it? We didn't want them to be able to prevent it because it completely breaks the menus if it just keeps trying to focus the sub menu trigger every time it closes.

Reply:
Ok makes sense.